app/controllers/school_administrator_dashboard_controller.rb

```diff
@@ -0,0 +1,49 @@
+class SchoolAdministratorDashboardController < ApplicationController
+  include StudentsQueryHelper
+
+  before_action :set_school, :authorize_for_school
+
+  def show
+    active_students = students_for_dashboard(@school)
+
+    dashboard_students = active_students.map { |student| individual_student_dashboard_data(student) }
+
+    @serialized_data = {
+      absences: dashboard_students.to_json
+    }
+    render 'shared/serialized_data'
+  end
+
+  private
+
+  def set_school
+    @school = School.find_by_slug(params[:id]) || School.find_by_id(params[:id])
+
+    redirect_to root_url if @school.nil?
+  end
+
+  def list_student_absences(student)
+    student.absences.map {|absence| absence.order(occurred_at: :desc)}
+  end
+
+  def individual_student_dashboard_data(student)
+    HashWithIndifferentAccess.new({
+      first_name: student.first_name,
+      last_name: student.last_name,
+      homeroom: student.try(:homeroom).try(:name),
+      absences: student.absences.order(occurred_at: :desc)
+      }
+    )
+  end
+
+  def students_for_dashboard(school)
+    school.students.active
+  end
+
+  def authorize_for_school
+    unless current_educator.is_authorized_for_school(@school)
```

Comment:
@edavidsonsawyer Like I mentioned in our email thread, this is going to block educators who are districtwide from viewing the graphs.

Reply:
I'm confused about this. Are districtwide educators not authorized for their schools? My understanding was the existing permissions are too broad and we want to restrict to principals and assistant principals.

Reply:
You're totally correct @edavidsonsawyer and I was wrong about this — `is_authorized_for_school` checks for `districtwide_access` first and returns `true` for districtwide educators, who aren't assigned to school. I double-checked to see who wrote that method and turns out I did lol.

Reply:
In terms of permissions being too broad, correct. @snoopyuri mentioned restricting to principals and assistant principals. While Insights doesn't know exactly who's a principal/AP based on the data we get from X2, checking for `current_educator.admin == true` should be an excellent proxy. I sent around a list of educators where `admin == true` called "insights_admins_for_review.txt" so @snoopyuri can review.